frontend/src/component/ui/Table.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component, ReactChild, ReactText } from 'react';
+
+import './Table.scss';
+
+interface IProps {
+  headings?: ReactChild[];
+  rows?: ReactChild[][];
+}
+
+export class Table extends Component<IProps> {
+  private constructHeadIfExists(headings: ReactChild[] | undefined) {
+    if (!headings || headings.length === 0) {
+      return null;
+    }
+    return <tr>{this.constructRow(headings, true)}</tr>;
+  }
+
+  private constructRow(row: ReactChild[], isHeader: boolean) {
+    return row.map((cell: ReactChild, cellIndex: number) => {
+      return isHeader
+        ? this.constructHeadingCell(cellIndex, cell)
+        : this.constructBodyCell(cellIndex, cell);
+    });
+  }
+
+  private constructHeadingCell(key: ReactText, content: ReactChild) {
+    return (
+      <th key={`${key}`} className="table-cell">
+        {content}
+      </th>
+    );
+  }
+
+  private constructBodyCell(key: ReactText, content: ReactChild) {
+    return (
+      <td key={`${key}`} className="table-cell">
+        {content}
+      </td>
+    );
+  }
+
+  private constructBodyIfExists(rows: ReactChild[][] | undefined) {
+    if (!rows || rows.length === 0) {
+      return null;
+    }
```

Comment:
```suggestion
    if (!rows) {
      return null;
    }
    
    if (rows.length === 0) {
        return null;
    }
```

Reply:
Any particular reason why you find this is better?

Reply:
@rohithbalaji123 I found it simpler to consider one condition at a time.

Reply:
Not a big deal.